app/cu_editor.py

```diff
@@ -396,11 +396,49 @@ def setTextBuffer(self, textBuffer):
       KEY_F2: textBuffer.bookmarkNext,
       KEY_F3: textBuffer.findAgain,
       KEY_F4: self.changeToPaletteWindow,
+      KEY_F5: self.changeToPopup,
       KEY_SHIFT_F2: textBuffer.bookmarkPrior,
       KEY_SHIFT_F3: textBuffer.findBack,
     })
     self.commandSet = commandSet
 
+class PopupController(app.controller.Controller):
+  """
+  A controller for pop up boxes to notify the user.
+  """
+  def __init__(self, view):
+    app.controller.Controller.__init__(self, view, 'popup')
+    self.callerSemaphore = None
+    def noOp(ch, meta):
+      app.log.info('noOp in PopupController')
+    self.commandDefault = noOp
+    self.commandSet = {
+      KEY_ESCAPE: self.changeToInputWindow,
+    }
+
+  def changeToInputWindow(self):
+    self.findAndChangeTo('inputWindow')
+    if self.callerSemaphore:
+      self.callerSemaphore.release()
+      self.callerSemaphore = None
+
+  def setOptions(self, options):
+    """
+    This function is used to change the options that are displayed in the
+    popup window as well as their functions.
+
+    Args:
+      options (dict): A dictionary mapping keys (ints) to its
+                      corresponding action.
+
+    Returns;
+      None.
+    """
+    self.commandSet = options
```

Comment:
Maybe make a note that the caller should include esc to close (since this will replace the entry from line 416); or maybe do .update(options) rather than assignment?

Reply:
Will add the note. I thought about the .update, but then the user may forget to overwrite or delete any old keys. I think it might be better to have them list it all out again when they modify it. Though, I guess we could have both an `update` and a `set` function.